File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
         ArrayList<Integer> result = new ArrayList<Integer>();

        for (int i = 0; i < flag.length; i++) {
            if (flag[i]) {
                for (int j = 0; j < arr[i]*2; j++) {
                    result.add(arr[i]);
                }
            } else {
                for (int j = 0; j < arr[i]; j++) {
                        result.remove(result.size()-1);
                }
            }
        }

        int answer[] = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            answer[i] = result.get(i);
        }
        return answer;
    }
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {
    var answer = [];
     for (let i = 0; i < flag.length; i++) {
        if (flag[i]) {
            for (let j = 0; j < arr[i] * 2; j++) {
                answer.push(arr[i]);
            }
        } else {
            for (let j = 0; j < arr[i]; j++) {
                answer.splice(-1, 1); 
            }
        }
    }

    return answer;
}